wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -722,7 +722,11 @@ def timesince_last_update(
             time_str = last_update.strftime("%H:%M")
 
         time_prefix = f"{time_prefix} " if time_prefix else time_prefix
-        by_user = f" by {user_display_name}" if user_display_name else user_display_name
+        by_user = ""
+        if user_display_name:
+            by_user = " " + _("by %(user_display_name)s") % {
+                "user_display_name": user_display_name
+            }
```

Comment:
Here's my thoughts - what if we avoid the clunky `by_user` completely here. Instead let's allow the translator to determine the sentence structure of the time by user and it's spaces. 

If we assume the time is at the start of this label we are probably making a bad assumption, same with assuming there is the 'by' before the name of the thing. 

Instead we should just skip the whole problem by generating a new translation with the sentence of the two parts. The time and the user. 

However, I agree with the concept of trailing/ leading whitespace should not be something that the translator needs to thin about. But usually that means we are doing something odd by building bits of a sentence standalone - which is probably the more critical problem. 

Here is a suggestion. 

Let me know and you think. 


```python
        if user_display_name:
           return _("%(time) by %(user_display_name)s") % {"time": f"{time_prefix}{time_str}", "user_display_name": user_display_name}

        return f"{time_prefix}{time_str}"
```

Reply:
The time variable should probably have a better name. Like `time_since`? So it's easier to understand that it's a time ago type time.

Reply:
I like your suggestion, and I thought about storing `f"{time_prefix}{time_str}"` in a new `time_since` variable and use it either in the gettext call or for the simple `return`.
But first I wanted to search for the usage of `time_prefix`, since I couldn't understand what's used there.
I found that there are two places in the templates where such a "time_prefix" is used ([here](https://github.com/wagtail/wagtail/blob/main/wagtail/contrib/modeladmin/templates/modeladmin/includes/header_with_history.html#L11) and [here](https://github.com/wagtail/wagtail/blob/main/wagtail/snippets/templates/wagtailsnippets/snippets/_header_with_history.html#L11)), and in both cases it is the untranslated string "at". I think we have to solve this problem too.

Maybe we replace "time_prefix" with a boolean flag (name?) and if set, we also use a single translation string, so this will result in additional gettext strings. Boolean, because I've only found the usage of the prefix "at". What do you think?

Reply:
yeah good call - passing in a single word like that is never really going to work well.

I like your approach. 

Please update accordingly and ping me on the PR and I can take another look. 

Thanks for this.

Reply:
Sry @lb-, there is one more thing :sweat_smile: 
After looking at `timesince_last_update()` again, I think there is another issue:
Currently the `user_display_name` is only used, if the passed date is from the same day, otherwise only the short version is returned. `user_display_name` is used quite often and I don't think that it is really intended, that the username is only shown when the date is from the current day.

If you're ok with that, I would first improve the tests to also contain cases where `user_display_name` is set and the date is older than one day (those would fail atm), then I would remove the calls to `timesince_simple()` from within `timesince_last_update()` and replace them all with separate translation strings, based on the given parameters.

Reply:
Maybe we move this issue to the 4.1 release to avoid blocking 4.0.3? These changes will create new translation strings, so it depends on the translations from transifex, and it is not really a blocker(?)

Reply:
I don't know if I really understand your last sentence, but since this issue has been moved to 4.1, I would now tackle all the i18n issues with `timesince_last_update()` as described in https://github.com/wagtail/wagtail/pull/9261#discussion_r985674134 and update this PR accordingly. I hope that this is also ok for you?

Reply:
Thanks. Sorry for the confusing last sentence above. 

I mean that if we changed things for 4.0 branch that created/removed translations and then did it again for 4.1 it would be frustrating for translators. 

However, now that this will be in 4.1 - it's not an issue.